source/adapters/level_zero/queue.cpp

```diff
@@ -1338,6 +1338,17 @@ ur_queue_handle_t_::executeCommandList(ur_command_list_ptr_t CommandList,
   }
 
   if (!UsingImmCmdLists) {
+    int numCompleted = 0;
+    for (auto &Event : CommandList->second.EventList) {
+      if (Event->ZeEvent) {
+        ze_result_t ZeResult =
+            ZE_CALL_NOCHECK(zeEventQueryStatus, (Event->ZeEvent));
+        if (ZeResult == ZE_RESULT_SUCCESS)
```

Comment:
to speed up this check, you can reverse this logic and exit given one event not completed. That way the if check later is not needed.

Reply:
querying status of events is expenisve operation, why is this needed?

Reply:
why on execute we check for event status ? This doesn't seem to be related operations.

Reply:
Going to add in the check for cb events and only query the status of counting events.